src/services/ramp-hooks.js

```diff
@@ -1343,3 +1345,249 @@ export const useAnnotations = () => {
+   * and there are multiple annotations that falls within the same time range.
+   */
+  useEffect(() => {
+    inPlayerRange ? setIsActive(true) : setIsActive(false);
+    if (autoScrollEnabled && inPlayerRange) {
+      autoScroll(annotationRef.current, containerRef, true);
+    }
+  }, [inPlayerRange]);
+
+  /**
+   * Truncate annotation text based on the width of the element on the page.
+   * Use a ResizeObserver to re-calculate truncated texts based on Annotations
+   * container re-size events
+   */
+  useEffect(() => {
+    const textBlock = annotationTextsRef.current;
+    let canvas, observer;
+    const calcTruncatedText = () => {
+      if (textBlock && texts?.length > 0) {
+        const textBlockWidth = textBlock.clientWidth;
+        const fontSize = parseFloat(getComputedStyle(textBlock).fontSize);
+        if (!isNaN(fontSize)) {
+          // Create a temporary canvas element to measure average character width
+          canvas = document.createElement('canvas');
+          const context = canvas.getContext('2d');
+          context.font = getComputedStyle(textBlock).font;
+
+          // Calculate average character width based on the specified font in CSS
+          const textWidth = context.measureText(texts).width;
+          const avgCharWidth = textWidth / texts.length;
+
+          // Calculate maximum number of characters that can be shown on avg character width
+          const charsPerLine = textBlockWidth / avgCharWidth;
+
+          /**
+           * To account for spaces at the end of line breaks, calculate max character for
+           * half a line width less than given MAX_LINES count
+           */
+          const maxCharactersToShow = charsPerLine * (MAX_LINES - 1)
+            + Math.floor(charsPerLine / 2);
+
+          let elementText = texts;
+
+          /**
+           * When texts has line breaks with shorter text in each line, pad each shorter line 
+           * until the length of it reaches the calculated charsPerLine number
+           */
+          if (texts.includes('<br>')) {
+            const lines = texts.split('<br>');
+            let paddedText = [];
+            for (let i = 0; i < lines.length; i++) {
+              const line = lines[i];
+              if (line.length < charsPerLine) {
+                // Account for the space for <br> for line breaks
+                const maxLineLength = charsPerLine > 4 ? charsPerLine - 4 : 0;
+                paddedText.push(line.padEnd(maxLineLength));
+              } else {
+                // Do nothing if text length is longer than charsPerLine
+                paddedText.push(line);
+              }
+            }
+            elementText = paddedText.join('<br>');
+          }
+
+          // Truncate text if the annotation text is longer than max character count
+          const { truncated, hasShowMore } = truncateText(elementText, maxCharactersToShow);
+          if (hasShowMore) {
```

Comment:
Is this check needed given it is happening on line 1471 outside of this function?
If it is needed what do you think about moving this check to the beginning of the function?  I think that would skip unnecessary processing if `hasShowMore` is `false`.

Reply:
The check on line 1471 is to check whether show more/less is enabled in the props, while this check is to see whether the text is truncated or not. And based on this check show more button is displayed next to the truncated text.
I'm not clear on how this could be moved up, can you please clarify?

Reply:
Nevermind!  I got confused with the variable names `hasShowMore`  vs.`enableShowMore`.

Reply:
Maybe I can change the variable names to make it clearer? 🙂

Reply:
I think it is makes sense as they are.  I just missed that they were different.